qiling/os/memory.py

```diff
@@ -63,6 +63,43 @@ def __read_string(self, addr: int) -> str:
     def __write_string(self, addr: int, s: str, encoding: str):
         self.write(addr, bytes(s, encoding) + b'\x00')
 
+    def __getitem__(self, key: Union[slice, int]) -> bytearray:
+        if isinstance(key, slice):
+            start = key.start
+            stop = key.stop
+            step = key.step
+
+            if step is not None and step != 1:
+                # step != 1 means we have to do copy, don't allow it.
+                raise IndexError("Only support slicing continous memory")
+
+            return self.ql.mem.read(start, max(0, stop - start))
+        elif isinstance(key, int):
+            return self.ql.mem.read(key, 1)[0]
+        else:
+            raise KeyError("Wrong type of key")
+    
+    def __setitem__(self, key: Union[slice, int], value: bytes):
+        if isinstance(key, int):
+            self.ql.mem.write(key, bytes([value]))
+        elif isinstance(key, slice):
+            start = key.start
+            stop = key.stop
+            step = key.step
+
+            if step is not None and step != 1:
+                raise IndexError("Only support slicing continous memory")
+            
+            if start is None:
+                raise IndexError("The start of memory is not supplied")
+            
+            if len(value) > stop - start:
```

Comment:
What happens when data length is less than slice length? I thought you are going to throw an excpetion if sizes do not match (i.e. when `len(value) != (stop - start)`)

Reply:
What about not enforcing the stop value? Where slice `stop` is not specified, it would mean either "end of memory" (or a "don't care", if you don't like the idea of spanning till the end of memory)

For example (note the missing stop value):
```python
ql.mem[addr:] = b'hello'
```
Pesonally I would prefer leaving the `int` key behavior to be replacing a single byte in the specified location, because that way the behavior of something like `ql.mem[addr] = data` is not ambiguous. Requiring the user to specify a slice with an empty `stop` would make it clearer. All that being said, both approaches make sense and acceptable.

Reply:
This looks great and intuitive!

Reply:
Done